### DISCUSSION TITLE
Graph to/from pandas.DataFrame

### DISCUSSION THREAD

**Discussion by clstaudt** [networkit/networkit#697] [Ideas]
Pandas is the work horse for data analysis in Python, so it would be good to have functions like

`from_dataframe(df, from_col, to_col, directed=False, weight_col=None, ...)`

`to_dataframe(G, from_col, to_col, weight_col=None, ...)`

where `df` is a `pandas.DataFrame`, `from_col` and `to_col` are columns of node ids, and every row specifies an edge.

**Comment by avdgrinten** [comment]
Hi Christian, I would welcome such functionality but OTOH there are little resources on our side to implement it, especially considering that we do not use Pandas and this request is not challenging from an algorithmic point of view. Would you be willing to provide a patch?

**Comment by hridaydutta123** [comment]
@clstaudt Are you working on it? I can take it up and work on the patch.

**Comment by clstaudt** [comment]
> @clstaudt Are you working on it? I can take it up and work on the patch.

@hridaydutta123 By all means, go ahead. Thanks!

**Comment by bielefem** [comment]
Since this Issue seems to have no progress.
I would try to write a Patch.

But first off all i would like to specify this feature a bit more.
Therefore i need more pieces of information on how you want would use it @clstaudt

because i am a little bit confused by
````python
from_dataframe(df, from_col, to_col, directed=False, weight_col=None, ...)

to_dataframe(G, from_col, to_col, weight_col=None, ...)
# where df is a pandas.DataFrame, from_col and to_col are columns of node ids, and every row specifies an edge.
````
in this case its not clear to my why need two different collums of ids

If we look at old versions of for example Netwrokx 

in the old version they just used [to_pandas_dataframe](https://networkx.github.io/documentation/networkx-1.10/reference/generated/networkx.convert_matrix.to_pandas_dataframe.html#networkx.convert_matrix.to_pandas_dataframe)
and nowerdays they have two functions [to_pandas_adjacency](https://networkx.github.io/documentation/stable/reference/generated/networkx.convert_matrix.to_pandas_adjacency.html) and [to_pandas_edgelist](https://networkx.github.io/documentation/stable/reference/generated/networkx.convert_matrix.to_pandas_edgelist.html#networkx.convert_matrix.to_pandas_edgelist)

which are APIs that only use the Graph and 1 list of node ids.

**Comment by avdgrinten** [comment]
I think the intent of the proposed `from_col`, `to_col` matches NetworkX's`source` and `target` in `to_pandas_edgelist`, i.e., it's just the name of the columns that specify source and target nodes of edges.
